gateleen-hook/src/main/java/org/swisspush/gateleen/hook/Route.java

```diff
@@ -51,6 +54,15 @@
     private Forwarder forwarder;
     private HttpClient selfClient;
 
+    static {
+        String keepAliveTimeoutProperty =  System.getProperty(HTTP_CONNECTION_KEEP_ALIVE_TIMEOUT);
```

Comment:
To be a bit more consistent in the usage of system properties you could introduce the `Map<String, Objec>` properties map. This is already used in several other Gateleen classes like Handlers.

For example in Crush, we also have the following code in the `MainVerticle`, which allows to define such properties based on environment:
```java
final Map<String, Object> props;
try {
	props = new ResourcePropertySource("classpath:/" + SERVER_NAME + "/config/environments/default-env.properties").getSource();

	Resource envResource = new ClassPathResource("/" + SERVER_NAME + "/config/environments/" + env + "-env.properties");
	if (envResource.exists()) {
		props.putAll(new ResourcePropertySource(envResource).getSource());
		log.info("Merging environment config from classpath for " + env);
	} else {
		log.info("No environment config in classpath for " + env);
	}

	String externalConfig = System.getProperty(CONFIG_DIR_SYSTEM_PROPERTY) + "/nemo/config.properties";

	Resource externalConfigResource = new FileSystemResource(externalConfig);

	if (externalConfigResource.exists()) {
		props.putAll(new ResourcePropertySource(externalConfigResource).getSource());
		log.info("Merging external config " + externalConfig);
	} else {
		log.info("No external config found under " + externalConfig);
	}
} catch (IOException e) {
	throw new RuntimeException(e);
}
```

By defining the `keepAliveTimeout` here, it would be used for all routes? Wouldn't it be better to allow the `keepAliveTimeout` in the configuration resource of routes and rules? So we could define it just for the routes/rules we want. With the global approach we could even define a global keepAlive and override it with specific configuration on route and rule basis.

I would suggest to adopt the json schema for routes and rules to allow the configuration of a `keepAliveTimeout` value

Reply:
In our case we can easily define system properties on a per stage base outside of some gateleen mechanism. What we have here is totally sufficient.

We actually want to use the keepAliveTimeout for dynamically created routes, we do not want to set in on per statically configured route in the rules resource.

Reply:
You added the possibility to define the keepAliveTimeout on routing rules but it is only usable for dynamically created routes. Even if you don't need it right now, I would still add it also for statically configured routes/rules. Besides that, it's a small effort and others could use it someday. I also think its not correct do have configurations on rules which you cannot use everywhere.

Reply:
Hmm, yeah, OK. HTTP/1.1 will probably stay with us a little bit longer, so there is a chance. Will add it.